File: 14. longestCommonPrefix.cpp
class Solution {
public:
	string longestCommonPrefix(vector<string>& strs) {
		string result = "";

		if (strs.size() == 0) return result;
		if (strs.size() == 1) return strs[0];

		bool toBreak = false;
		for ( int i = 0; i < strs[0].length() && toBreak == false; i++ )
		{
			char ch = strs[0].at(i);
			for (int j = 1; j < strs.size(); j++)
			{
				if (strs[j].length() > i)
				{
					if (strs[j].at(i) != ch)
					{
						toBreak = true;
						break;
					}
				}
				else
				{
					toBreak = true;
					break;
				}
			}

			if (toBreak == false)
			{
				result.append(1, ch);
			}
		}

		return result;
	}
};

File: 14. Longest Common Prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length == 0) return new String();
        String first = strs[0];
        StringBuilder lcp = new StringBuilder("");
        for (int i = 0; i < first.length(); i++) {
            char ch1 = first.charAt(i);
            boolean append = true;
            for (int j = 1; j < strs.length; j++) {
                if (strs[j].length() <= i) {
                    append = false;
                    break;
                }

                char ch2 = strs[j].charAt(i);
                if (ch1 != ch2) {
                    append = false;
                    break;
                }
            }
            if (append) {
                lcp.append(ch1);
            } else {
                break;
            }
        }
        return lcp.toString();
    }
}